docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,984 @@
+
+If you paste the following query into the left side of the window and press the play button you should see data for posts with their IDs and descriptions returned:
+
+```graphql
+query {
+  posts {
+    id
+    description
+  }
+}
+```
+
+This data is an example of the data you will source with your plugin.
+
+_You can also see a running version of the GraphQL playground associated with a distinct API at [https://gatsby-source-plugin-api.glitch.me/](https://gatsby-source-plugin-api.glitch.me/), which is running the `api` folder in a Glitch project, like you would when you run `npm start` on your own computer._
+
+#### Plugin behavior
+
+Your plugin will have the following behavior:
+
+- Make an API request to the demo API
+- Convert the data in the API response to Gatsby's node system
+- Link the nodes together so you can query for an author on each post
+- Accept plugin options to customize how your plugin works
+- Optimize images from Unsplash URLs so they can be used with `gatsby-image`
+
+### Setup projects for plugin development
+
+You'll need to setup an example site and create a plugin inside it to begin building.
+
+#### Setup an example site
+
+Create a new Gatsby site with the `gatsby new` command, based on the hello world starter.
+
+```shell
+gatsby new example-site https://github.com/gatsbyjs/gatsby-starter-hello-world
+```
+
+This site generated by the `new` command is where the plugin will be installed, giving you a place to test the code for your plugin.
+
+#### Setup a source plugin
+
+Create a new Gatsby plugin with the `gatsby new` command, this time based on the plugin starter.
+
+```shell
+gatsby new source-plugin https://github.com/gatsbyjs/gatsby-starter-plugin
+```
+
+This will create your plugin in a separate project than your example site, but you could also include it in your [site's `plugins` folder](/docs/loading-plugins-from-your-local-plugins-folder/).
+
+Your plugin starts with a few files from the starter, which can be seen in the snippet below:
+
+```text
+/example-site
+/source-plugin
+├── .gitignore
+├── gatsby-browser.js
+├── gatsby-node.js
+├── gatsby-ssr.js
+├── index.js
+├── package.json
+└── README.md
+```
+
+The file where all the logic will live is in the **`gatsby-node.js`**. This file is where Gatsby expects to find any usage of the [Gatsby Node APIs](https://www.gatsbyjs.org/docs/node-apis/). These allow customization/extension of default Gatsby settings affecting pieces of the site build process. This file is where all the logic for sourcing data will live.
+
+#### Install your plugin in the example site
+
+You need to install your plugin in the site to be able to test that your code is running. Gatsby only knows to run plugins that are included in its `gatsby-config.js` file. Open up the `gatsby-config.js` file in the `example-site` and [add your plugin using `require.resolve`](/docs/creating-a-local-plugin/#using-requireresolve-and-a-filepath).
```

Comment:
That's a great point, will add a blurb about that 👍 

```suggestion
You need to install your plugin in the site to be able to test that your code is running. Gatsby only knows to run plugins that are included in its `gatsby-config.js` file. Open up the `gatsby-config.js` file in the `example-site` and [add your plugin using `require.resolve`](/docs/creating-a-local-plugin/#using-requireresolve-and-a-filepath). If you decide to publish your plugin it can be installed with an `npm install <plugin-name>` and including the name of the plugin in the config instead of `require.resolve`.
```